src/Toolkit/Toolkit.WPF/UI/Controls/PopupViewer/MediaPopupElementView.cs

```diff
@@ -0,0 +1,156 @@
+    /// </summary>
+    public class MediaPopupElementView : Control
+    {
+        private ButtonBase? _previousButton;
+        private ButtonBase? _nextButton;
+        private int selectedIndex = 0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaPopupElementView"/> class.
+        /// </summary>
+        public MediaPopupElementView()
+        {
+            DefaultStyleKey= typeof(MediaPopupElementView);
+        }
+        
+        /// <inheritdoc />
+        public override void OnApplyTemplate()
+        {
+            if (_previousButton != null)
+            {
+                _previousButton.Click -= OnPreviousButtonClicked;
+            }
+            if (_nextButton != null)
+            {
+                _nextButton.Click -= OnNextButtonClicked;
+            }
+            _previousButton = GetTemplateChild("PreviousButton") as ButtonBase;
+            if (_previousButton != null)
+            {
+                _previousButton.Click += OnPreviousButtonClicked;
+            }
+            _nextButton = GetTemplateChild("NextButton") as ButtonBase;
+            if (_nextButton != null)
+            {
+                _nextButton.Click += OnNextButtonClicked;
+            }
+            UpdateContent();
+            base.OnApplyTemplate();
+        }
+
+        /// <summary>
+        /// Gets or sets the currently display <see cref="PopupMedia"/>.
+        /// </summary>
+        public PopupMedia? CurrentItem
+        {
+            get => GetValue(CurrentItemProperty) as PopupMedia;
+            set => SetValue(CurrentItemProperty, value);
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="CurrentItem"/> dependency property.
+        /// </summary>       
+        public static readonly DependencyProperty CurrentItemProperty =
+            DependencyProperty.Register(nameof(CurrentItem), typeof(PopupMedia), typeof(MediaPopupElementView), new PropertyMetadata(null));
+
+        private void UpdateContent()
+        {
+            var itemCount = Element?.Media?.Count ?? 0;
+            if (_previousButton != null)
+                _previousButton.Visibility = itemCount < 2 ? Visibility.Collapsed : Visibility.Visible;
+            if (_nextButton != null)
+                _nextButton.Visibility = itemCount < 2 ? Visibility.Collapsed : Visibility.Visible;
+            PopupMedia? content = null;
+            if (Element?.Media != null)
+            {
+                if(selectedIndex >= 0 && selectedIndex < itemCount)
+                {
+                    content = Element.Media[selectedIndex];
+                }
+            }
+            CurrentItem = content;
+        }
+
+        private void OnPreviousButtonClicked(object sender, RoutedEventArgs e)
+        {
+            selectedIndex--;
+            if (selectedIndex < 0)
+            {
+                selectedIndex = Element?.Media?.Count ?? 1 - 1;
```

Comment:
This line would benefit from parens to clarify operator precedence
```suggestion
                selectedIndex = (Element?.Media?.Count ?? 1) - 1;
```